tests/test_brisque.py

```diff
@@ -3,96 +3,87 @@
+def test_brisque_raises_if_wrong_reduction(prediction_grey: torch.Tensor, device: Any) -> None:
     for mode in ['mean', 'sum', 'none']:
-        brisque(prediction_grey, reduction=mode)
+        brisque(prediction_grey.to(device), reduction=mode)
 
     for mode in [None, 'n', 2]:
         with pytest.raises(KeyError):
-            brisque(prediction_grey, reduction=mode)
+            brisque(prediction_grey.to(device), reduction=mode)
 
 
-def test_brisque_values_grey(prediction_grey: torch.Tensor) -> None:
-    score = brisque(prediction_grey, reduction='none', data_range=1.)
-    score_baseline = torch.tensor([BRISQUE().get_score((img * 255).type(torch.uint8).squeeze().numpy())
-                                   for img in prediction_grey])
-    assert torch.isclose(score, score_baseline, atol=1e-1, rtol=1e-3).all(), f'Expected values to be equal to ' \
-                                                                             f'baseline prediction.' \
-                                                                             f'got {score} and {score_baseline}'
+def test_brisque_values_grey(device: Any) -> None:
+    img = np.array(Image.open('tests/assets/goldhill.gif'))
+    prediction_grey = torch.tensor(img).unsqueeze(0).unsqueeze(0)
+    score = brisque(prediction_grey.to(device), reduction='none', data_range=255)
+    score_baseline = BRISQUE().get_score(img)
+    assert torch.isclose(score, torch.tensor(score_baseline).to(score), rtol=1e-3), \
+        f'Expected values to be equal to baseline prediction, got {score.item()} and {score_baseline}'
 
 
-def test_brisque_values_rgb(prediction_rgb) -> None:
-    score = brisque(prediction_rgb, reduction='none', data_range=1.)
-    score_baseline = [BRISQUE().get_score((img * 255).type(torch.uint8).squeeze().permute(1, 2, 0).numpy()[..., ::-1])
-                      for img in prediction_rgb]
-    assert torch.isclose(score,
-                         torch.tensor(score_baseline),
-                         atol=1e-1, rtol=1e-3).all(), f'Expected values to be equal to ' \
-                                                      f'baseline prediction.' \
-                                                      f'got {score} and {score_baseline}'
+def test_brisque_values_rgb(device: Any) -> None:
+    img = np.array(Image.open('tests/assets/I01.BMP'))
+    prediction_rgb = (torch.tensor(img, dtype=torch.float32).permute(2, 0, 1).unsqueeze(0))
+    score = brisque(prediction_rgb.to(device), reduction='none', data_range=255.)
+    score_baseline = BRISQUE().get_score(prediction_rgb[0].permute(1, 2, 0).numpy()[..., ::-1])
+    assert torch.isclose(score, torch.tensor(score_baseline).to(score), rtol=1e-3), \
+        f'Expected values to be equal to baseline prediction, got {score.item()} and {score_baseline}'
 
 
-def test_brisque_all_zeros_or_ones() -> None:
-    size = (1, 1, 256, 256)
+def test_brisque_all_zeros(device: Any) -> None:
+    size = (1, 1, 128, 128)
```

Comment:
Test name is misleading, as you also check tensor of ones inside. 
Consider doing this using `pytest.mark.parametrize`

```suggestion
@pytest.mark.parametrize(
    "input,expectation",
    [(torch.zeros(2, 1, 96, 96), AssertionError),
      (torch.ones(2, 1, 96, 96), AssertionError)],
)
def test_brisque_for_special_cases(input: torch.Tensor, expectation: Any, device: str) -> None:
    with expectation:
        brisque(input.to(device), reduction='mean')

```